ci/da-ghc-lib/compile.yml

```diff
@@ -7,11 +7,12 @@ parameters:
 jobs:
 - job: ${{ parameters.final_job_name }}
   pool:
-    vmImage: 'Ubuntu 16.04'
+    name: ubuntu_20_04
+    demands: assignment -equals default
   variables:
-    ghc-lib-sha: '77fcc42eef2256ba9bb61a7e442a65bedb19a35d'
+    ghc-lib-sha: '42e5c306dcfbc84b83336fdd531023e93bfcc5b2'
     base-sha: '9c787d4d24f2b515934c8503ee2bbd7cfac4da20'
-    patches: '323ed5bbd2af0fc8bad7debb1c2ca2a050206bde 833ca63be2ab14871874ccb6974921e8952802e9'
+    patches: '25a18b7a5b3ea4dfadbcd9fd55410c9fbc2b8e28 833ca63be2ab14871874ccb6974921e8952802e9'
```

Comment:
See https://github.com/digital-asset/ghc/pull/66 will bump once that is approved & merged